docs(gpu): add concepts (#2013)
scaleway/docs-content: compute/gpu/concepts.mdx

<Concept opened>

> ## AI-Enabled applications

Reviewer: ```suggestion
## AI-Enabled Applications
```

Reviewer 2: @jsudraud If **AI-Enabled applications** is not the name of a product/feature, I'd keep it lowercase.

Reviewer: Yes, you're right. I thought it was a title, but it's not

Reviewer 3: moi je mettrais AI-enabled applications

Reviewer 4: idem

Author: ```suggestion
## AI-enabled applications
```

> The proliferation of AI-Enabled applications and innovative use cases is poised to bring transformative changes to various domains, including business, social interactions, and human-machine interfaces.

Reviewer 5: ```suggestion
The proliferation of AI-enabled applications and innovative use cases is poised to bring transformative changes to various domains, including business, social interactions, and human-machine interfaces.
```

> These applications will feature conversational and multimodal user interfaces (UIs) that will revolutionize interactions within smart spaces, smart robots, and autonomous vehicles.

Reviewer 5: I'd put this bit first, since it actually defines what they are, and then "The proliferation of..." after

</Concept>

<Concept opened>

## Automatic Mixed Precision

> In the realm of deep neural network training, the standard approach has been to use the IEEE single-precision format.

Reviewer 5: Since it's a concepts page, I'd start with the definition rather than the context. Maybe you can reorder things a bit?

> Nevertheless, an alternative method called mixed precision has emerged, allowing training with half precision while retaining the network's accuracy achieved with single precision.

Reviewer: ```suggestion
Nevertheless, an alternative method called mixed precision has emerged, allowing training with half-precision while retaining the network's accuracy achieved with single precision.
```

This innovative technique, known as mixed-precision, combines both single- and half-precision representations.

> Major AI frameworks (such as TensorFlow and PyTorch) have embraced this approach, offering automatic mixed precision support to accelerate AI training on NVIDIA GPUs with minimal code adjustments.

Reviewer: ```suggestion
Major AI frameworks (such as TensorFlow and PyTorch) have embraced this approach, offering automatic mixed-precision support to accelerate AI training on NVIDIA GPUs with minimal code adjustments.
```

</Concept>

<Concept opened>

## BF16

BF16, or Brain Floating Point 16, is a numerical format optimized for neural network computations. 
It strikes a balance between range and precision, making it efficient for deep learning tasks while maintaining accuracy.
BF16 uses 1 bit for the sign, 8 bits for the exponent, and 7 bits for the mantissa (fractional part). This format provides a wider range than IEEE 754 half-precision (FP16) while still maintaining better precision than integer formats. 
It was introduced to address the limitations of FP16 for deep learning tasks, offering improved efficiency and accuracy for training and running neural network models.

</Concept>

<Concept>

## CUDA

> CUDA is a parallel computing platform and API model created by NVIDIA. CUDA is an acronym for **c**ompute **u**nified **d**evice **a**rchitecture.

Reviewer 3: ```suggestion
CUDA is a parallel computing platform and API model created by NVIDIA. CUDA is an acronym for **C**ompute **U**nified **D**evice **A**rchitecture.
```

</Concept>

<Concept>

## Data-Centric AI

Data-Centric AI delves into diverse data and analytics techniques, aiming to derive insights from business, Internet of Things (IoT), and sensor data.
Additionally, it seeks to enhance AI-enabled decisions, making them not only more accurate but also explainable and ethical.
In this context, the trustworthiness of AI systems, which operate with varying degrees of autonomy, is essential, and their risks need to be effectively managed.

</Concept>

<Concept>

> ## Docker

Reviewer 3: https://www.scaleway.com/en/docs/compute/gpu/concepts/

On a deja la definition de Docker ici

Scaleway provides a number of pre-built Docker images, which allow you to [run a Docker container on your GPU Instance](/compute/gpu/how-to/use-gpu-with-docker/) and access a pre-installed Pipenv environment ideal for your AI projects.

</Concept>

<Concept>

## Flexpoint

Flexpoint is a numerical representation format tailored for efficient computations in machine learning and neural networks.
It offers a dynamic range and adaptable precision, optimizing both accuracy and computational efficiency.
This format's flexibility in bit allocation makes it well-suited for tasks where balancing precision and performance is crucial.

</Concept>
<Concept>

## Generative AI

Generative AI involves AI methods that learn representations from data and utilize them to create entirely new artifacts while retaining similarities to the original data.
These artifacts can be applied for both positive and negative purposes.
Generative AI has the ability to generate fresh media content, such as text, images, videos, and audio, as well as synthetic data and models of physical objects.
Additionally, it finds applications in fields like drug discovery and material design.

</Concept>

<Concept>

## GPU

> **G**raphical **P**rocessing **U**nit (GPU) became a go-to term for the specialized electronic circuits, designed to power graphics on a machine. The term was popularized in the late 1990s by the chip manufacturer NVIDIA. GPUs were originally produced primarily to drive high-quality gaming experiences, producing life-like digital graphics. Today, those capabilities are being harnessed more broadly for data processing, articificial intelligence, rendering and video encoding.

Reviewer 2: ```suggestion
**G**raphical **P**rocessing **U**nit (GPU) became a go-to term for the specialized electronic circuits, designed to power graphics on a machine. The term was popularized in the late 1990s by the chip manufacturer NVIDIA. GPUs were originally produced primarily to drive high-quality gaming experiences, producing life-like digital graphics. Today, those capabilities are being harnessed more broadly for data processing, artificial intelligence, rendering and video encoding.
```

</Concept>

<Concept>

> ## Model-Centric AI

Reviewer 5: ```suggestion
## Model-centric AI
```

> Model-Centric AI focuses on the most promising and emerging techniques that will pave the way for future groundbreaking advancements in the field of AI.

Reviewer 5: ```suggestion
Model-centric AI focuses on the most promising and emerging techniques that will pave the way for future groundbreaking advancements in the field of AI.
```

These techniques include foundation models, composite AI, physics-informed AI, neuromorphic computing, and biology-inspired algorithms.

> By leveraging these innovative approaches, Model-Centric AI aims to drive revolutionary progress in the world of artificial intelligence.

Reviewer 3: ```suggestion
By leveraging these innovative approaches, Model-centric AI aims to drive revolutionary progress in the world of artificial intelligence.
```

</Concept>

<Concept>

## Inference and Training

At a high level, working with deep neural networks involves a two-stage process.
First, the neural network undergoes training, where its parameters are determined by using labeled examples of inputs and corresponding desired outputs.
Then, the trained network is deployed for inference, utilizing the learned parameters to classify, recognize, and process unfamiliar inputs.

</Concept>

<Concept>

## Moore's Law

Moore's law, originally observed by Intel co-founder Gordon Moore in 1965, describes a consistent trend wherein the number of transistors per square inch on integrated circuits doubles every year since their inception.

> This observation led to the prediction that this trend would persist into the foreseeable future.

Reviewer 4: Not sure this part is useful to understand the concept

Pipenv is pre-installed on all of Scaleway's AI Docker images for GPU Instances, making it easy to [use virtual environments](/compute/gpu/how-to/use-pipenv/). Pipenv replaces Anaconda for this purpose.

</Concept>

<Concept>

## PyTorch

PyTorch is a GPU-accelerated tensor computational framework with a Python front end.
It provides a seamless integration with popular Python libraries like NumPy, SciPy, and Cython, allowing for easy extension of its functionality.

</Concept>

<Concept>

## Responsible and Human-Centric AI

> Responsible and Human-Centric AI emphasizes the positive impact of AI on individuals and society, while also addressing the need to manage and mitigate AI-related risks.

Reviewer 3: ```suggestion
Responsible and Human-centric AI emphasizes the positive impact of AI on individuals and society, while also addressing the need to manage and mitigate AI-related risks.
```

It encourages vendors to adopt ethical and responsible practices in their AI implementations.
Additionally, it advocates for combining AI with a human touch and common sense to ensure a more holistic and beneficial AI experience for everyone.

</Concept>

<Concept>

> ## Scratch Storage

Reviewer 5: ```suggestion
## Scratch storage
```

Scratch storage refers to a type of temporary storage space used in computing systems, particularly in high-performance computing (HPC) environments and supercomputers. 
It is used for short-term storage of data that is being actively processed, but it is not meant for long-term or permanent storage. 
It is an ephemeral storage solution that does not support features like snapshots, backups, or restores.

</Concept>

<Concept>

> ## Structural Sparsity

Reviewer 5: ```suggestion
## Structural sparsity
```

> Modern AI networks are characterized by their substantial size, containing millions or even billions of parameters.

Reviewer 4: ```suggestion
Modern AI networks are characterized by their substantial size, as they contain millions or even billions of parameters.
```
